tinyiiod.h

```diff
@@ -22,44 +22,55 @@
 #include <stddef.h>
 #include <stdint.h>
 #include <stdlib.h>
+#include <sys/types.h>
 
 #if !defined(__ssize_t_defined) && !defined(_SSIZE_T_DEFINED)
-typedef long int ssize_t;
 #define __ssize_t_defined
 #define _SSIZE_T_DEFINED
 #endif
 
 struct tinyiiod;
 
 struct tinyiiod_ops {
+	int32_t (*read)(int32_t *instance_id, char *buf, size_t len);
 	/* Read from the input stream */
-	void (*read)(char *buf, size_t len);
+	int32_t (*read_line)(int32_t *instance_id, char *buf);
```

Comment:
this looks like an API change
i am not sure how spread/used this code is;
but, if changing API, we may want to start a larger discussion about it;

in any case, these should be as part of separate commits, and have some more details about them